contracts/SpeedMarkets/SpeedMarketsAMM.sol

```diff
@@ -0,0 +1,518 @@
+    }
+
+    function createNewMarketWithDifferentCollateral(
+        bytes32 asset,
+        uint64 strikeTime,
+        SpeedMarket.Direction direction,
+        bytes[] calldata priceUpdateData,
+        address collateral,
+        uint collateralAmount,
+        bool isEth
+    ) external payable nonReentrant notPaused {
+        require(multicollateralEnabled, "Multicollateral onramp not enabled");
+        uint buyinAmount = _convertCollateral(collateral, collateralAmount, isEth);
+        _createNewMarket(asset, strikeTime, direction, buyinAmount, priceUpdateData, false);
+    }
+
+    function createNewMarketWithDifferentCollateralAndDelta(
+        bytes32 asset,
+        uint64 delta,
+        SpeedMarket.Direction direction,
+        bytes[] calldata priceUpdateData,
+        address collateral,
+        uint collateralAmount,
+        bool isEth
+    ) external payable nonReentrant notPaused {
+        require(multicollateralEnabled, "Multicollateral onramp not enabled");
+        uint buyinAmount = _convertCollateral(collateral, collateralAmount, isEth);
+        _createNewMarket(asset, uint64(block.timestamp + delta), direction, buyinAmount, priceUpdateData, false);
+    }
+
+    function _convertCollateral(
+        address collateral,
+        uint collateralAmount,
+        bool isEth
+    ) internal returns (uint buyinAmount) {
+        uint convertedAmount;
+        if (isEth) {
+            convertedAmount = multiCollateralOnOffRamp.onrampWithEth{value: msg.value}(collateralAmount);
+        } else {
+            IERC20Upgradeable(collateral).safeTransferFrom(msg.sender, address(this), collateralAmount);
+            IERC20Upgradeable(collateral).approve(address(multiCollateralOnOffRamp), collateralAmount);
+            convertedAmount = multiCollateralOnOffRamp.onramp(collateral, collateralAmount);
+        }
+        buyinAmount = (convertedAmount * (ONE - safeBoxImpact - lpFee)) / ONE;
+    }
+
+    function _createNewMarket(
+        bytes32 asset,
+        uint64 strikeTime,
+        SpeedMarket.Direction direction,
+        uint buyinAmount,
+        bytes[] memory priceUpdateData,
+        bool transferSusd
+    ) internal {
+        require(supportedAsset[asset], "Asset is not supported");
+        require(buyinAmount >= minBuyinAmount && buyinAmount <= maxBuyinAmount, "wrong buy in amount");
+        require(
+            strikeTime >= (block.timestamp + minimalTimeToMaturity),
+            "time has to be in the future + minimalTimeToMaturity"
+        );
+        require(strikeTime <= block.timestamp + maximalTimeToMaturity, "time too far into the future");
+
+        uint fee = pyth.getUpdateFee(priceUpdateData);
+        pyth.updatePriceFeeds{value: fee}(priceUpdateData);
+
+        PythStructs.Price memory price = pyth.getPrice(assetToPythId[asset]);
+
+        require((price.publishTime + maximumPriceDelay) > block.timestamp, "Stale price");
```

Comment:
I would just add `price.publishTime + maximumPriceDelay) > block.timestamp && price.price > 0`
Better to be safe, in case they have a glitch with 0 value

Reply:
added